src/api/pdc_client_connect.c

```diff
@@ -7454,6 +7409,68 @@ PDC_Client_query_kvtag_col(const pdc_kvtag_t *kvtag, int *n_res, uint64_t **pdc_
     FUNC_LEAVE(ret_value);
 }
 
+// Delete a tag specified by a name, and whether it is from a container or an object
+static perr_t
+PDCtag_delete(pdcid_t obj_id, char *tag_name, int is_cont)
+{
+    perr_t                         ret_value = SUCCEED;
+    hg_return_t                    hg_ret    = 0;
+    uint64_t                       meta_id;
+    uint32_t                       server_id;
+    hg_handle_t                    metadata_del_kvtag_handle;
+    metadata_get_kvtag_in_t        in;
+    struct _pdc_obj_info *         obj_prop;
+    struct _pdc_cont_info *        cont_prop;
+    struct _pdc_client_lookup_args lookup_args;
+
+    FUNC_ENTER(NULL);
+
+    if (is_cont) {
+        cont_prop = PDC_cont_get_info(obj_id);
+        meta_id   = cont_prop->cont_info_pub->meta_id;
+    }
+    else {
+        obj_prop = PDC_obj_get_info(obj_id);
+        meta_id  = obj_prop->obj_info_pub->meta_id;
+    }
+
+    server_id = PDC_get_server_by_obj_id(meta_id, pdc_server_num_g);
+
+    debug_server_id_count[server_id]++;
+
+    if (PDC_Client_try_lookup_server(server_id, 0) != SUCCEED)
+        PGOTO_ERROR(FAIL, "==CLIENT[%d]: ERROR with PDC_Client_try_lookup_server", pdc_client_mpi_rank_g);
+
+    HG_Create(send_context_g, pdc_server_info_g[server_id].addr, metadata_del_kvtag_register_id_g,
+              &metadata_del_kvtag_handle);
+
+    // Fill input structure
+    in.obj_id = meta_id;
+
+    if (is_cont)
+        in.hash_value = PDC_get_hash_by_name(cont_prop->cont_info_pub->name);
+    else
+        in.hash_value = PDC_get_hash_by_name(obj_prop->obj_info_pub->name);
+    in.key = tag_name;
+
+    hg_ret = HG_Forward(metadata_del_kvtag_handle, metadata_add_tag_rpc_cb /*reuse*/, &lookup_args, &in);
```

Comment:
I would suggest we add more comments here to note that there is not much logic in the add_tag_rpc_cb function so that's why it can be reused here for a tag-delete operation...